### ISSUE TITLE
지도 학습 정리

### ISSUE THREAD

**Issue by Kimuksung** [Kimuksung/bigdata#25]
>지도 학습  
정의 : Label이 명확한 상태이고 올바른 정답이 있는 상태에서 학습 시키는 방법

-학습된 알고리즘을 이용하여 predict -> 결과값에 따라 category -> classfication continous -> regression

>비지도 학습
정의: Label이 명확한 상태이지만 올바른 정답이 없다.

-Clustring -> Hidden feature / structure

강화 학습
정의:state에 따라 행동하고 보상을 얻으면서 학습을 진행시킨다.

머신러닝: 통계학과 다르게 data로 부터 기반을 쌓는다.
classfication  : 훈련 data set을 목표 집단 까지 예측,평가 ( 고객 데이터 -> 컴퓨터 부속 구매)

clustering : 목표집단을 예측 하지 않으며, 유사한 특징을 통해 집단화

분류 알고리즘
Linear classfication -> logistic
kernerl -> k-nearst
decision tree -> decision tree / random forest / xgboost

Linear accuracy
-MSE = 평균 차이의 제곱
-RSquared = 높을 수록 적합한 model
-RMSE = 오차의 제곱
-confunsion matrix = table을 이용하여 predict와 실제 data 비교
-ROC
-overfitting = sample data로만 modeling 할 경우 data 정확도가 낮아진다.
해결방법: nomalization / crossvalidation / 더 많은 data

logistic regression
- independent 한 linear 특징을 이용한 통계 기법
- odss , logit function을 이용

decision tree
- 관측값과 목표값을 연결 시켜주는 model , 시각적이고 명시적
- entropy 를 사용하여 탐욕 알고리즘 ->  매 번 좋아 보이는 것 선택(splitting)
알고리즘 적으로 적합한 알고리즘이 아니기 때문에 쌓이다보면 문제가 발생
-가지치기 : 과적합 방지

앙상블
1) 배깅 - 동일한 알고리즘 N개 학습 => 정확성 , 안정성 (bias / variance 낮다)
병렬적으로 modeling하여 평균을 이용하여 나타난다.
random forest
2) 부스팅- 맞추기 어려운 문제를 풀기 위해 고안 , 순차 학습
•objective = "binary:logistic": we will train a binary classification model ;
"binary:logistic" : y변수 이항
•max_depth = 2: the trees won't be deep, because our case is very simple ;
tree 구조가 간단한 경우 : 2
•nthread = 2: the number of cpu threads we are going to use;
cpu 사용 수 : 2
•nrounds = 2: there will be two passes on the data, the second one will enhance the model by further reducing the difference between ground truth and prediction.
실제값과 예측값의 차이를 줄이기 위한 반복학습 횟수
•eta = 1 : eta control the learning rate
학습률을 제어하는 변수(Default: 0.3)
숫자가 낮을 수록 모델의 복잡도가 높아지고, 컴퓨팅 파워가 더많이 소요
부스팅 과정을보다 일반화하여 오버 피팅을 방지하는 데 사용
•verbose = 0 : no message
0이면 no message, 1이면 성능에 대한 정보 인쇄, 2이면 몇 가지 추가 정보 인쇄

**Comment by Kimuksung** [comment]
boosting
https://xgboost.readthedocs.io/en/latest/parameter.html
